applications/crossbar/src/modules/cb_functions.erl

```diff
@@ -0,0 +1,386 @@
+    _ = crossbar_bindings:bind(<<"*.content_types_accepted.functions">>, ?MODULE, 'content_types_accepted'),
+    _ = crossbar_bindings:bind(<<"*.validate.functions">>, ?MODULE, 'validate'),
+    _ = crossbar_bindings:bind(<<"*.billing">>, ?MODULE, 'billing'),
+    _ = crossbar_bindings:bind(<<"*.execute.get.functions">>, ?MODULE, 'get'),
+    _ = crossbar_bindings:bind(<<"*.execute.put.functions">>, ?MODULE, 'put'),
+    _ = crossbar_bindings:bind(<<"*.execute.post.functions">>, ?MODULE, 'post'),
+    _ = crossbar_bindings:bind(<<"*.execute.patch.functions">>, ?MODULE, 'patch'),
+    _ = crossbar_bindings:bind(<<"*.execute.delete.functions">>, ?MODULE, 'delete'),
+    _ = crossbar_bindings:bind(<<"*.finish_request">>, ?MODULE, 'finish_request'),
+    ok.
+
+%%------------------------------------------------------------------------------
+%% @doc Authenticates the incoming request, returning true and context
+%% if the requestor is known, or false if not.
+%% @end
+%%------------------------------------------------------------------------------
+-spec authenticate(cb_context:context()) ->
+                          {'true', cb_context:context()} |
+                          'false'.
+authenticate(Context) ->
+    authenticate(Context, cb_context:req_verb(Context), cb_context:req_nouns(Context)).
+
+-spec authenticate(cb_context:context(), http_method(), req_nouns()) ->
+                          {'true', cb_context:context()} |
+                          'false'.
+authenticate(Context, ?HTTP_GET, [{<<"functions">>, []}]) ->
+    {'true', Context};
+authenticate(Context, ?HTTP_GET, [{<<"functions">>, [?PATH_TOKEN_SAMPLES | _]}]) ->
+    {'true', Context};
+authenticate(Context, ?HTTP_PUT, [{<<"functions">>, _}]) ->
+    {'true', Context};
+authenticate(Context, ?HTTP_PATCH, [{<<"functions">>, _}]) ->
+    {'true', Context};
+authenticate(Context, ?HTTP_DELETE, [{<<"functions">>, _}]) ->
+    {'true', Context};
+authenticate(_Context, _Verb, _Nouns) -> 'false'.
+
+%%------------------------------------------------------------------------------
+%% @doc Authorizes the incoming request, returning true if the requestor is
+%% allowed to access the resource, or false if not.
+%% @end
+%%------------------------------------------------------------------------------
+-spec authorize(cb_context:context()) -> boolean().
+authorize(Context) ->
+    authorize(Context
+             ,cb_context:req_nouns(Context)
+             ,cb_context:req_verb(Context)
+             ).
+
+-spec authorize(cb_context:context(), req_nouns(), http_method()) -> boolean().
+authorize(Context, [{<<"functions">>, _}], ?HTTP_POST) ->
+    cb_context:is_superduper_admin(Context)
+        orelse cb_context:is_account_admin(Context);
+authorize(_Context, [{<<"functions">>, [?PATH_TOKEN_SAMPLES | _]}], ?HTTP_GET) ->
+    'true';
+authorize(_Context, [{<<"functions">>, []}], ?HTTP_GET) ->
```

Comment:
Still not sure (looking at functions.md) what `GET /v2/functions` should return?

Reply:
This will return a list functions under the specified account. I will update the function.md for clarification. An example would be 
```
curl -X GET -H "X-Auth-Token: $AUTH_TOKEN" http://localhost:8000/v2/accounts/155caa38cd30097b9be07294d4110306/functions/ 
{
    "auth_token": "eyJhbGc....",
    "data": [
        {
            "enabled": true,
            "function_js": "if (cfDoc&&cfDoc.CallerNumber&&CallerNumber===\"1000\") {return {\"module\":\"voicemail\",\"children\": {}, \"data\": { \"action\": \"compose\", \"callerid_match_login\": false, \"id\": \"afee84a507643901833f839fc494307d\", \"interdigit_    timeout\": 2000, \"max_message_length\": 500, \"single_mailbox_login\": false }}}; else return {\"module\": \"do_not_disturb\", \"data\": {\"action\": \"activate\"},\"children\":{\"_\":{\"module\":\"hangup\", \"data\":{}}}};",
            "id": "d5c8895baadf1a8a2b4badd94b9c89fb",
            "name": "reply_to_caller_vm"
        },
        {
            "enabled": true,
            "function_js": "return {\"originalDoc\": cfDoc}",
            "id": "ac8dc25d6d7212729868bd2330f5be93",
            "name": "New Function"
        }
    ],
    "node": "aklmvc3b_Vsnfxg2qz619g",
    "page_size": 2,
    "request_id": "0687fa4405df1f4a18de3145748b5af5",
    "revision": "2b885097e8780d158c0ff748649f4905",
    "status": "success",
    "timestamp": "2019-08-08T20:05:18Z",
    "tokens": {
        "consumed": 1,
        "remaining": 100
    },
    "version": "4.3.1"
}```

Reply:
removed. it will return empty [].